deckhouse-controller/pkg/registry/deckhouse.go

```diff
@@ -0,0 +1,230 @@
+// Copyright 2022 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/ettle/strcase"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	regTransport "github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"gopkg.in/yaml.v2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/deckhouse/deckhouse/deckhouse-controller/pkg/controller/module-controllers/utils"
+	"github.com/deckhouse/deckhouse/go_lib/dependency"
+	"github.com/deckhouse/deckhouse/go_lib/dependency/cr"
+	"github.com/deckhouse/deckhouse/go_lib/libapi"
+)
+
+type DeckhouseService struct {
```

Comment:
I don't think we have to make it exportable also as `NewDeckhouseService` constructor. It's not planned to use it in the other packages at the moment

Reply:
we have a linter rule to not return not public structures.
if it's ok - we can rename it.

Reply:
done